dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/database/migration/V9_02_02__DML_Fill_urlaliases.java

```diff
@@ -0,0 +1,285 @@
+          } catch (JsonProcessingException e) {
+            throw new RuntimeException("Cannot parse " + w + ": " + e, e);
+          }
+        });
+    LOGGER.info("Successfully added {} UrlAliases for {}", identifiables.size(), tableName);
+  }
+
+  private UrlAlias buildUrlAlias(
+      JdbcTemplate jdbcTemplate,
+      EntityType entityType,
+      IdentifiableType identifiableType,
+      UUID uuid,
+      String language,
+      String label,
+      UUID websiteUuid) {
+    Locale locale = Locale.forLanguageTag(language);
+    String baseSlug = slugGenerator.generateSlug(label);
+    String slug = baseSlug;
+    int suffix = 0;
+    while (true) {
+      try {
+        if (!hasUrlAlias(jdbcTemplate, slug, uuid, locale, websiteUuid)) break;
+      } catch (SQLException e) {
+        throw new RuntimeException(
+            "Cannot check availability of URLAlias for uuid="
+                + uuid
+                + ", locale="
+                + locale
+                + ", slug="
+                + slug
+                + ": "
+                + e,
+            e);
+      }
+      suffix++;
+      slug = String.format("%s-%d", baseSlug, suffix);
+    }
+
+    if (!slug.equals(baseSlug)) {
+      LOGGER.warn("Building slug with suffix: " + slug);
+    }
+
+    UrlAlias urlAlias = new UrlAlias();
+    urlAlias.setUuid(UUID.randomUUID());
+    urlAlias.setTargetLanguage(locale);
+    urlAlias.setPrimary(true);
+    urlAlias.setTargetUuid(uuid);
+    urlAlias.setTargetIdentifiableType(identifiableType);
+    urlAlias.setTargetEntityType(entityType);
+    urlAlias.setSlug(slug);
+    if (websiteUuid != null) {
+      Website website = new Website();
+      website.setUuid(websiteUuid);
+      urlAlias.setWebsite(website);
+    }
+
+    return urlAlias;
+  }
+
+  private void saveUrlAlias(JdbcTemplate jdbcTemplate, UrlAlias urlAlias) throws SQLException {
+    String updateQuery =
+        "insert into url_aliases (uuid,created,last_published,\"primary\",slug,target_entity_type,target_identifiable_type,target_language,target_uuid,website_uuid) VALUES(?::uuid,CURRENT_TIMESTAMP,CURRENT_TIMESTAMP,?,?,?,?,?,?::uuid,?::uuid);";
+    jdbcTemplate.update(
+        updateQuery,
+        urlAlias.getUuid().toString(),
+        urlAlias.isPrimary(),
+        urlAlias.getSlug(),
+        urlAlias.getTargetEntityType() != null ? urlAlias.getTargetEntityType().toString() : null,
+        urlAlias.getTargetIdentifiableType().toString(),
+        urlAlias.getTargetLanguage().toString(),
+        urlAlias.getTargetUuid().toString(),
+        urlAlias.getWebsite() != null ? urlAlias.getWebsite().getUuid().toString() : null);
+  }
+
+  private boolean hasUrlAlias(
+      JdbcTemplate jdbcTemplate, String slug, UUID uuid, Locale locale, UUID websiteUuid)
+      throws SQLException {
+
+    if (websiteUuid == null) {
+      return jdbcTemplate.queryForInt(
+              "SELECT count(*) from url_aliases where slug=? and target_uuid=uuid(?) and target_language=?",
```

Comment:
```suggestion
              "SELECT count(*) from url_aliases where website_uuid is null and slug=? and target_uuid=uuid(?) and target_language=?",
```
Müsste hier nicht die Website explizit `null` sein?